crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs

```diff
@@ -132,10 +132,28 @@ impl Analyzer<'_, '_> {
             .with_child(ScopeKind::Flow, true_facts_for_rhs.clone(), |child: &mut Analyzer| -> VResult<_> {
                 child.ctx.should_store_truthy_for_access = false;
 
+                let mut type_param = vec![];
```

Comment:
I think this is wrong. I'll fix it

Reply:
I tried fixing https://github.com/dudykr/stc/blob/e28482355dc0684d07189869e8368187637642b7/crates/stc_ts_file_analyzer/src/analyzer/expr/misc.rs#L44-L47 as

```rust
        let type_args = e.type_args.validate_with(self)?;

        let ty = e
            .expr
            .validate_with_args(self, (mode, Some(&type_args), type_ann))
            .context("tried to verify expr of ts instantiation expression")?;

        let new = self.expand_generics_with_type_args(e.span, ty, &type_args)?;

        Ok(new)
```

but there were some regressions, so I gave up fixing it with this PR. I'll create one for it.

Reply:
e.type_args is not eq type_args...
Therefore, the following strange results were derived.

Reply:
first I will merged main